storage/innobase/buf/buf0flu.cc

```diff
@@ -389,21 +381,15 @@ void buf_page_write_complete(const IORequest &request, bool error)
   mysql_mutex_assert_not_owner(&buf_pool.mutex);
   mysql_mutex_assert_not_owner(&buf_pool.flush_list_mutex);
 
-  if (request.is_LRU())
+  if (bpage->oldest_modification() == 2)
```

Comment:
I personally find the literal numeric constants easier to read than named constants, such as `CLEAN=0`, `CLEAN_IN_FLUSH_LIST=1`, `TEMPORARY=2`. A debugger would show the numeric values `0`, `1`, `2`, and while reading the code in a debugging session it could only obfuscate things if you do not remember that `CLEAN_IN_FLUSH_LIST` actually means 1.

In any case, should this be changed, it should be done independently of this fix.